hazelcast/src/test/java/com/hazelcast/client/connectionstrategy/ConfiguredBehaviourTest.java

```diff
@@ -113,6 +114,19 @@ public void testAsyncStartTrue() {
         });
     }
 
+    @Test
+    public void testAsyncStartTrueShouldNotBlock_whenThereIsClientStateToSend() {
+        class A { }
+        ClientConfig clientConfig = new ClientConfig();
+        clientConfig.getConnectionStrategyConfig().getConnectionRetryConfig().setClusterConnectTimeoutMillis(-1);
+        ClientUserCodeDeploymentConfig clientUserCodeDeploymentConfig = new ClientUserCodeDeploymentConfig();
+        clientUserCodeDeploymentConfig.addClass(A.class);
+        clientUserCodeDeploymentConfig.setEnabled(true);
+        clientConfig.setUserCodeDeploymentConfig(clientUserCodeDeploymentConfig);
+        clientConfig.getConnectionStrategyConfig().setAsyncStart(true);
+        hazelcastFactory.newHazelcastClient(clientConfig);
+    }
```

Comment:
can we also verify that after the client connects to the cluster, the state is in fact sent?

Reply:
Makes sense, let me use the logic in https://github.com/hazelcast/hazelcast/blob/7c3894b1474dba56648cf10ad8441615486fcdea/hazelcast/src/test/java/com/hazelcast/client/usercodedeployment/ClientUserCodeDeploymentTest.java#L204